app/code/Magento/ExtendedOrder/composer.json

```diff
@@ -0,0 +1,24 @@
+{
+    "name": "magento/module-extended-order",
```

Comment:
Maybe something in the lines of `InventoryInStorePickupOrderExtension`? In this case even name would be self-explanatory.